server/redis.ts

```diff
@@ -18,12 +19,36 @@ const options = {
         }
       : undefined,
 };
-const client = new Redis(process.env.REDIS_URL, options);
-const subscriber = new Redis(process.env.REDIS_URL, options);
-// More than the default of 10 listeners is expected for the amount of queues
-// we're running. Increase the max here to prevent a warning in the console:
-// https://github.com/OptimalBits/bull/issues/1192
-client.setMaxListeners(100);
-subscriber.setMaxListeners(100);
-
-export { client, subscriber };
+
+export default class RedisAdapter extends Redis {
+  constructor(url: string | undefined) {
+    if (!(url || "").startsWith("ioredis://")) {
+      super(process.env.REDIS_URL, defaultOptions);
+    } else {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const decodedString = Buffer.from(url!.slice(10), "base64").toString();
+      const customOptions = JSON.parse(decodedString);
```

Comment:
We should try/catch around this with a useful error.

Reply:
In my view this alert is a false positive, so there is no error to catch here. The if clause directly above this code will catch the null case (which is a perfectly fine way to initialize ioredis with the default options). It is a possible way to initialize the redis class in the current master so I wanted to preserve this behavior. Am I missing something here?

Reply:
On second though, you are probably talking about everything else that can go wrong here except the error that is explicitly ignored ^^
I will add the try/catch.

Reply:
Yea, someone will add invalid json and then complain that it breaks in a weird way :)